### PR TITLE
Performance/canvas selection updates canvas first

### LINKED ISSUE #719: Implement High Priority Selection State in the canvas controls
#716 seems like a winner, perception-wise

### PR DESCRIPTION
**Pull request by balazsbajorics**
Fixes #719

**Problem:**
The actions dispatched when selecting a View on the canvas take a long time. 

**Fix:**
We can fix the perceptual speed of the editor by first drawing the new selection controls _then_ dispatching the actions in the next frame.

**Commit Details:**
- renamed `NewCanvasControlsClass` to `NewCanvasControlsInner` because it's not even a class anymore
- added `usePropControlledStateV2` which does not render twice if the prop changes.
- added `localHighlightedViews` and `localSelectedViews` to `NewCanvasControlsInner` using `usePropControlledStateV2`
- added a new callback `setSelectedViewsLocally` which is called from `select-mode-control-container`
- `select-mode-control-container`'s `selectComponent` only calls the dispatch after _two_ animation frames are fired (the first one is the current one, the second one is the next one relative to the event.)

**Notes:**
This only works for mouse-based selection on the canvas. Keyboard-based selection and selection in the navigator is still slow as ever.

### CODE REVIEW

File: editor/src/components/canvas/controls/select-mode-control-container.tsx [hunk 1]
```diff
@@ -104,13 +106,28 @@ export class SelectModeControlContainer extends React.Component<
...
       }
-      let selectActions = [
-        EditorActions.clearHighlightedViews(),
-        EditorActions.selectComponents(updatedSelection, false),
-        EditorActions.setLeftMenuTab(LeftMenuTab.UINavigate),
-        EditorActions.setRightMenuTab(RightMenuTab.Inspector),
-      ]
-      this.props.dispatch(selectActions, 'canvas')
+
+      ;(ReactDOM as any).flushSync(() => {
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/select-mode-control-container.tsx:110]: If this is secret functionality could you add a comment indicating what this is up to please?

**Review by seanparsons — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

File: editor/src/components/canvas/controls/select-mode-control-container.tsx (hunk 1 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/controls/select-mode-control-container.tsx:110]: good idea!

Comment (balazsbajorics) [reply editor/src/components/canvas/controls/select-mode-control-container.tsx]: I removed this altogether instead of explaining it :)

**Review by Rheeseyb — COMMENTED** [review]
:+1:

**Pull request merged**